internal/request.go

```diff
@@ -53,20 +52,20 @@ func (s *SPOA) processRequest(msg spoe.Message) ([]spoe.Action, error) {
 				return nil, fmt.Errorf("application %q not found", arg.Value.(string))
 			}
 		case "id":
-			tx = app.waf.NewTransaction(context.Background())
-			tx.ID, ok = arg.Value.(string)
+			id, ok := arg.Value.(string)
 			if !ok {
 				return nil, fmt.Errorf("invalid argument for http request id, string expected, got %v", arg.Value)
 			}
-			tx.Variables.UniqueID.Set(tx.ID)
+			tx := app.waf.NewTransactionWithID(id)
```

Comment:
Don't redeclare the var, scope should come from line 38, otherwise it will throw nil pointer exceptions.